src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -503,7 +503,22 @@ private function createTypeNotFoundMessageCallback(TypedReference $reference, st
 
     private function createTypeNotFoundMessage(TypedReference $reference, string $label, string $currentId): string
     {
-        if (!$r = $this->container->getReflectionClass($type = $reference->getType(), false)) {
+        $type = $reference->getType();
+
+        if ($this->container->hasDefinition($type) && $this->container->getDefinition($type)->hasTag('container.excluded')) {
+            return sprintf('Cannot autowire service "%s": %s has type "%s" but this symbol has been excluded.', $currentId, $label, $type);
```

Comment:
Minor comment: the word `symbol` in this error message is a bit confusing to me.

I looked at a real error message in the tests you added:

```
Cannot autowire service "a": argument "$k" of method
"Symfony\Component\DependencyInjection\Tests\Compiler\NotGuessableArgument::__construct()"
has type "Symfony\Component\DependencyInjection\Tests\Compiler\Foo" but this symbol has been excluded.
```

If this is my original code in this case:

```php
use Symfony\Component\DependencyInjection\Tests\Compiler\Foo;

class NotGuessableArgument
{
    public function __construct(Foo $a)
    {
        // ...
    }
}
```

I'm not sure I could understand what's happening and what could I change to fix this error.

Reply:
symbol = class or interface. We can add some logic do decide for one or the other. Would that fix your concern?

Reply:
Yes, if others agree I'd prefer to not see the "symbol" word in this error message.

But, the error message might need more tweaks. In summary it says: _"We cannot autowire XXX because YYY has been excluded"_.

What I'm missing is: who excluded YYY? Why this did happen? Can I do something to fix it? If yes, what can I do?

Maybe there's no need to answer all questions, but I'd definitely need more information/help if I tried to fix issue this myself.

But maybe this is an obscure error message that nobody will see in real apps and we can ignore this. Thanks.

Reply:
"excluded" refers to the "exclude" entry in the services.yaml file:
https://github.com/symfony/recipes/blob/8af9d2e95d0bf5792bd19ba7c4916fb43aebe1ee/symfony/framework-bundle/5.4/config/services.yaml#L18

Any idea to make this message more actionable?

Reply:
maybe `has been excluded from loading` could make the message clearer

Reply:
Updated, WDYT?
`Cannot autowire service "Foo": it needs an instance of "App\\Entity\\Post" but this type has been excluded in "config/services.yaml".`

Reply:
Yes! Now I can understand the issue perfectly. Thanks a lot for extra effort to tweak this message!